### PR TITLE
[Android] Simplify / fix modechange

### PR DESCRIPTION
**Pull request by ghost**
## Description
This PR simplifies modechange implementation by splitting SetHDMIState into 2 methods:
- InitiateModeChange which triggers the delay timer
- SetHDMIState which handles HDMI_AUDIO_PLUG intents.

## Motivation and Context
OnResetDisplay was not fired on Android TV Box when it woke up after sleep, therefore Audio Sink was not properly initialized.

## How Has This Been Tested?
Start Kodi on ShieldTV , wait until OS went into sleep, wake up, play a movie.

## Types of change
- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

### CODE REVIEW

File: xbmc/windowing/android/WinSystemAndroid.cpp [hunk 1]
```diff
@@ -219,66 +217,48 @@ void CWinSystemAndroid::UpdateResolutions(bool bUpdateDesktopRes)
...
+  if (connected)
   {
-    if (m_dispResetState & RESET_WAIT_HDMIPLUG)
+    if (m_dispResetTimer->IsRunning())
     {
-      m_dispResetState &= ~RESET_WAIT_HDMIPLUG;
-
-      if (m_dispResetTimer->GetElapsedMilliseconds() >= delay)
-        // Most probably a pseudo (2 sec) timer -> signal now
+      if (m_dispResetTimer->GetElapsedMilliseconds() >=
```
Comment (fritsch) [inline xbmc/windowing/android/WinSystemAndroid.cpp]: A comment here would be good that this is used for the "wakeup check", e.g. we exit here and stop the timer when it was just "misused" to wait for the initial hotplug enable.

File: xbmc/windowing/android/WinSystemAndroid.cpp [hunk 2]
```diff
@@ -219,66 +217,48 @@ void CWinSystemAndroid::UpdateResolutions(bool bUpdateDesktopRes)
...
     }
-    if (m_dispResetState == RESET_WAIT_TIMER)
-    {
-      for (auto resource : m_resources)
-        resource->OnResetDisplay();
-      m_dispResetState = 0;
-      m_dispResetTimer->Stop();
-    }
+
+    for (auto resource : m_resources)
```
Comment (fritsch) [inline xbmc/windowing/android/WinSystemAndroid.cpp]: The lock could be moved above this statement IF you don't use it as well to cover the m_dispResetTimer, etc.

File: xbmc/windowing/android/WinSystemAndroid.cpp [hunk 3]
```diff
@@ -219,66 +217,48 @@ void CWinSystemAndroid::UpdateResolutions(bool bUpdateDesktopRes)
...
-    CSingleLock lock(m_resourceSection);
-    if (m_dispResetState & RESET_WAIT_HDMIPLUG)
-    {
-      // Let HDMI_PLUG trigger the reset
-      m_dispResetState &= ~RESET_WAIT_HDMIPLUG;
-      return;
-    }
-  }
-  SetHDMIState(HDMI_STATE_CONNECTED);
+  SetHDMIState(true);
```
Comment (fritsch) [inline xbmc/windowing/android/WinSystemAndroid.cpp]: I am not sure until today, if this thread, which executes the OnTimeout has additional side-effects. Other platforms reset that stuff on their mainloop. On the other hand you properly take the resource lock below and it should work.

**Review by fritsch — COMMENTED** [review]
Please check my comments, nothing critical - in general this should work as is.

File: xbmc/windowing/android/WinSystemAndroid.cpp [hunk 4]
```diff
@@ -219,66 +215,52 @@ void CWinSystemAndroid::UpdateResolutions(bool bUpdateDesktopRes)
...
+  if (connected)
   {
-    if (m_dispResetState & RESET_WAIT_HDMIPLUG)
+    if (m_dispResetTimer->IsRunning())
     {
-      m_dispResetState &= ~RESET_WAIT_HDMIPLUG;
-
-      if (m_dispResetTimer->GetElapsedMilliseconds() >= delay)
-        // Most probably a pseudo (2 sec) timer -> signal now
+      // We stop the timer if OS supports HDMI_AUDIO_PLUG intend
```
Comment (garbear) [inline xbmc/windowing/android/WinSystemAndroid.cpp]: intend -> intent

File: cmake/addons/bootstrap/repositories/binary-addons.txt [hunk 5]
```diff
@@ -1 +1 @@
-binary-addons https://github.com/xbmc/repo-binary-addons.git Matrix
+binary-addons https://github.com/peak3d/repo-binary-addons.git Matrix
```
Comment (Rechi) [inline cmake/addons/bootstrap/repositories/binary-addons.txt]: ?

**Comment by fritsch** [comment]
jenkins build this please

**Comment by peak3d** [comment]
UWP build failure (still) not relevant.

**Pull request merged**